src/coreComponents/mesh/generators/VTKMeshGenerator.cpp

```diff
@@ -148,101 +159,180 @@ loadVTKMesh( Path const & filePath )
+    {
+      extractor->SetCellIds( cells.dataIfContiguous(), LvArray::integerConversion< vtkIdType >( cells.size() ) );
+      extractor->Update();
+
+      vtkNew< vtkUnstructuredGrid > ug;
+      ug->ShallowCopy( extractor->GetOutputDataObject( 0 ) );
+      result->SetPartition( LvArray::integerConversion< unsigned int >( p ), ug );
+    }
+  }
+  return result;
+}
 
-/**
- * @brief Redistribute the mesh among the available MPI ranks
- * @details this method will also generate global ids for points and cells in the VTK Mesh
- * @param[in] loadedMesh the mesh that was loaded on one or several MPI ranks
- * @param[out] cuts the bounding boxes used by the VTK partitioner needed to compute neighboring ranks
- */
 vtkSmartPointer< vtkUnstructuredGrid >
-redistributeMesh( vtkUnstructuredGrid & loadedMesh,
-                  std::vector< vtkBoundingBox > & cuts )
+redistributeByCellGraph( vtkUnstructuredGrid & mesh,
+                         MPI_Comm const comm,
+                         int const numRefinements )
 {
-  // Generate global IDs for vertices and cells
-  vtkNew< vtkGenerateGlobalIds > generator;
-  generator->SetInputDataObject( &loadedMesh );
-  generator->Update();
+  GEOSX_MARK_FUNCTION;
 
-  // Redistribute data all over the available ranks
+  // Use int64_t here to match ParMETIS' idx_t
+  ArrayOfArrays< int64_t, int64_t > const elemToNodes = buildElemToNodes< int64_t >( mesh );
+  int64_t const numProcs = MpiWrapper::commSize( comm );
+  array1d< int64_t > const newParts = parmetis::partMeshKway( elemToNodes.toViewConst(), comm, 3, numRefinements );
+  vtkSmartPointer< vtkPartitionedDataSet > const splitMesh = splitMeshByPartition( mesh, numProcs, newParts.toViewConst() );
+  return vtk::redistribute( *splitMesh, MPI_COMM_GEOSX );
+}
+
+vtkSmartPointer< vtkUnstructuredGrid >
+redistributeByKdTree( vtkUnstructuredGrid & mesh )
+{
+  GEOSX_MARK_FUNCTION;
+
+  // Use a VTK filter which employs a kd-tree partition internally
   vtkNew< vtkRedistributeDataSetFilter > rdsf;
-  rdsf->SetInputDataObject( generator->GetOutputDataObject( 0 ) );
+  rdsf->SetInputDataObject( &mesh );
   rdsf->SetNumberOfPartitions( MpiWrapper::commSize() );
   rdsf->Update();
-
-  cuts = rdsf->GetCuts();
   return vtkUnstructuredGrid::SafeDownCast( rdsf->GetOutputDataObject( 0 ) );
 }
 
 /**
- * @brief Compute the potential rank neighbor list
- * @param[in] cuts the bounding boxes used by the VTK partitioner for all ranks
- * @details Compute the rank neighbors. The assumption is that 2 ranks are neighbors if
- * the corresponding bounding boxes intersect.
+ * @brief Compute the rank neighbor candidate list.
+ * @param[in] boundingBoxes the bounding boxes used by the VTK partitioner for all ranks
+ * @return the list of neighboring MPI ranks, will be updated
  */
-std::unordered_set< int > computeMPINeighborRanks( std::vector< vtkBoundingBox > cuts )
+std::vector< int >
+findNeighborRanks( std::vector< vtkBoundingBox > boundingBoxes )
```

Comment:
With the latest commit of the PR, I confirm that the new approach to find the neighbor ranks works very well. Previously I had to disable the ghosting to reach the end of initialization, but now everything goes smoothly, the ghost cells seem to be exchanged without issue, and the simulation can proceed.